bokeh/plotting/helpers.py

```diff
@@ -426,6 +409,49 @@ def _process_active_tools(toolbar, tool_map, active_drag, active_scroll, active_
     else:
         raise ValueError("Got unknown %r for 'active_tap', which was not a string supplied in 'tools' argument" % active_tap)
 
+def _get_argspecs(glyphclass):
+    argspecs = OrderedDict()
+    for arg in glyphclass._args:
+        spec = {}
+        prop = getattr(glyphclass, arg)
+        spec['desc'] = " ".join(x.strip() for x in prop.__doc__.strip().split("\n\n")[0].split('\n'))
+        spec['default'] = prop.class_default(glyphclass)
+        spec['type'] = prop.__class__.__name__
+        argspecs[arg] = spec
+    return argspecs
```

Comment:
Perhaps this could go directly on `Glyph` classes? I'd feel better about adding a test to it there